crates/miden-tx/src/host/mod.rs

```diff
@@ -529,6 +552,74 @@ where
         Ok(())
     }
 
+    /// Checks if the necessary witness for accessing the map item is already in the merkle store,
+    /// and if not, extracts all necessary data for requesting it.
+    ///
+    /// Expected stack state: `[KEY, ROOT, index]`
+    pub fn on_account_storage_before_get_map_item(
+        process: &ProcessState,
+    ) -> Result<TransactionEventHandling, TransactionKernelError> {
+        let map_key = process.get_stack_word(0);
+        let map_root = process.get_stack_word(1);
+        let slot_index = process.get_stack_item(8);
+
+        Self::on_account_storage_before_get_or_set_map_item(slot_index, map_root, map_key, process)
+    }
+
+    /// Checks if the necessary witness for accessing the map item is already in the merkle store,
+    /// and if not, extracts all necessary data for requesting it.
+    ///
+    /// Expected stack state: `[index, KEY, NEW_VALUE, OLD_ROOT]`
+    pub fn on_account_storage_before_set_map_item(
+        process: &ProcessState,
+    ) -> Result<TransactionEventHandling, TransactionKernelError> {
+        let slot_index = process.get_stack_item(0);
+        let map_key = Word::from([
+            process.get_stack_item(4),
+            process.get_stack_item(3),
+            process.get_stack_item(2),
+            process.get_stack_item(1),
+        ]);
```

Comment:
Should there be a method in the processor crate to get a word at a specific offset? Feels like this pattern repeats a bunch